quil-rs/src/stats/execution_graph.rs

```diff
@@ -132,34 +133,46 @@ impl ExecutionGraph {
     }
 
     /// Returns the longest path from an initial instruction (one with no prerequisite instructions) to a final instruction (one with no dependent instructions).
-    pub fn gate_depth(&self) -> Result<usize, Error> {
-        let path_lengths = self.path_fold(
-            0,
-            |depth: usize, instruction: &Instruction| -> Result<usize, Error> {
-                if let Instruction::Gate(_) = instruction {
-                    Ok(depth + 1)
-                } else {
-                    Ok(depth)
-                }
-            },
-        )?;
-        Ok(path_lengths.into_iter().max().unwrap_or_default())
+    pub fn gate_depth(&self) -> usize {
+        let path_lengths = self
+            .path_fold(
+                0,
+                |depth: usize, instruction: &Instruction| -> Result<usize, Infallible> {
+                    if let Instruction::Gate(_) = instruction {
+                        Ok(depth + 1)
+                    } else {
+                        Ok(depth)
+                    }
+                },
+            )
+            .unwrap_or_else(|_| {
+                unreachable!(
+                    "'gate_depth' callback is infallible, so path_fold should not return an error"
+                )
+            });
```

Comment:
Is this functionally different from `expect`? (Also below)

Reply:
Hm, for some reason I thought it changed optimization, but no, only `unreachable_unchecked` does that, which makes sense.